tests/yam/agama/boot_agama.pm

```diff
@@ -63,6 +63,10 @@ sub run {
 
     $grub_menu->expect_is_shown();
     $grub_menu->edit_current_entry();
+    unless (get_var('AGAMA_AUTO')) {
+        my $profile_url = generate_json_profile();
+        push(@params, "agama.auto=\"$profile_url\"");
+    }
```

Comment:
this block should go under `if agama auto` above, so if it is x86_64 we will go with this and if not, the previous mechanism for the other architectures. It is preparation, doesn't need to to be in the middle of the ui code, anyway it runs in the workers, you don't need openqa api.